lib/ansible/modules/lineinfile.py

```diff
@@ -542,9 +585,10 @@ def main():
     path = params['path']
     firstmatch = params['firstmatch']
     regexp = params['regexp']
+    search_string = params['search_string']
```

Comment:
fwiw, most of the changes could have been simplified by simply doing:

```
if params['search_string']:
    regexp = re.escape(params['search_string'])
```

Then the majority of other changes would have been unnecessary, as you could have kept just using the regex paths, but with an escaped string that gets treated literally.

Reply:
That's excellent. I didn't know about `re.escape()`. I think that really gets to the heart of the original feature request.

Reply:
WOW Thanks! I go for it!

Reply:
@sivel @samdoran  I'm concerned about this:

https://docs.python.org/3/library/re.html#re.escape

I've been testing the solution and I prefer the current approach as it is more reliable and controllable.

Find the difference between 3.6 and 3.7 respectively:

```py
\<FilesMatch\ \"\.py\[45\]\?\$\"\>
```

```py
<FilesMatch\\ "\\.py\\[45\\]\\?\\$">

A replacement like this: `<FilesMatch ".py[45]?$">` is not working with this approach.

Thank you!!
```

Reply:
From this tests, I'm going to keep the current approach if you agree.